cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -621,6 +616,54 @@ mkConsensusTracers mbEKGDirect trSel verb tr nodeKern fStats = do
      when (isRollForward ev) $
        sendEKGDirectCounter ekgDirect "cardano.node.metrics.served.header.counter.int"
 
+traceBlockFetchServerMetrics
+  :: forall blk. ()
+  => Maybe EKGDirect
+  -> STM.TVar Int64
+  -> STM.TVar Int64
+  -> STM.TVar SlotNo
+  -> Tracer IO (TraceBlockFetchServerEvent blk)
+  -> Tracer IO (TraceBlockFetchServerEvent blk)
+traceBlockFetchServerMetrics Nothing _ _ _ tracer = tracer
+traceBlockFetchServerMetrics (Just ekgDirect) tBlocksServed tLocalUp tMaxSlotNo tracer = Tracer bsTracer
+
+  where
+    bsTracer :: TraceBlockFetchServerEvent blk -> IO ()
+    bsTracer e@(TraceBlockFetchServerSendBlock p) = do
+      traceWith tracer e
+      let mSlotNo = case pointSlot p of
+                         Origin -> Nothing
+                         At slotNo -> Just slotNo
+
+      (served, mbLocalUpstreamyness) <- atomically $ do
+          served <- STM.modifyReadTVar' tBlocksServed (+1)
+          maxSlotNo <- STM.readTVar tMaxSlotNo
+          case mSlotNo of
+               Nothing ->
+                   return (served, Nothing)
+               Just slotNo ->
+                   case compare maxSlotNo slotNo of
+                        LT -> do
+                            STM.writeTVar tMaxSlotNo slotNo
+                            lu <- STM.modifyReadTVar' tLocalUp (+1)
```

Comment:
What is `modifyReadTVar'`?

Is it equialvent to:
```
STM.modifyTVar' tLocalUP (+1) >> STM.readTVar tLocalUP
```

Reply:
https://github.com/input-output-hk/cardano-node/blob/aa7d41199227029cf80b1d5672c5401d1c73f57f/cardano-node/src/Cardano/Node/STM.hs#L24-L30